examples/c/ex_col_store.c

```diff
@@ -0,0 +1,519 @@
+            break;
+        case 5:
+            strcpy(w.day, "SAT");
+            break;
+        case 6:
+            strcpy(w.day, "SUN");
+            break;
+        default:
+            assert(false);
+        }
+        /* 24-hour-time 0-2400. */
+        w.hour = rand() % 2401;
+        /* Temperature range: 0-50C.  */
+        w.temp = rand() % 51;
+        /* Feels like temperature range 0-50C */
+        w.feels_like_temp = rand() % 51;
+        /* Humidity range: 0-100%. */
+        w.humidity = rand() % 101;
+        /* Pressure range: 900-1100pa */
+        w.pressure = (rand() % (1100 + 1 - 900)) + 900;
+        /* Wind range: 0-200 km/hr. */
+        w.wind = rand() % 201;
+        /* latitude: 0-180 degrees. */
+        w.loc_lat = rand() % 181;
+        /* longitude: 0-90 degrees. */
+        w.loc_long = rand() % 91;
+
+        country = rand() % 7;
+        switch (country) {
+        case 0:
+            strcpy(w.country, "AUS");
+            break;
+        case 1:
+            strcpy(w.country, "GBR");
+            break;
+        case 2:
+            strcpy(w.country, "USA");
+            break;
+        case 3:
+            strcpy(w.country, "NZD");
+            break;
+        case 4:
+            strcpy(w.country, "IND");
+            break;
+        case 5:
+            strcpy(w.country, "CHI");
+            break;
+        case 6:
+            strcpy(w.country, "RUS");
+            break;
+        default:
+            assert(false);
+        }
+
+        w_array[i] = w;
+    }
+}
+
+static int
+find_min_and_max_temp(
+  WT_SESSION *session, uint16_t start_time, uint16_t end_time, int *min_temp, int *max_temp)
+{
+    WT_CURSOR *end_time_cursor, *join_cursor, *start_time_cursor;
+    uint64_t recno;
+    int exact;
+    int ret;
+    uint16_t hour;
+    uint8_t temp;
+
+    /* Open cursors needed by the join. */
+    error_check(
+      session->open_cursor(session, "join:table:weather(hour,temp)", NULL, NULL, &join_cursor));
+    error_check(
+      session->open_cursor(session, "index:weather:hour", NULL, NULL, &start_time_cursor));
+    error_check(session->open_cursor(session, "index:weather:hour", NULL, NULL, &end_time_cursor));
+
+    /*
+     * Select values WHERE (hour >= start AND hour <= end). Find the starting record closest to
+     * desired start time.
+     */
+    start_time_cursor->set_key(start_time_cursor, start_time);
+    error_check(start_time_cursor->search_near(start_time_cursor, &exact));
+    ret = 0;
+    if (exact == -1)
+        ret = start_time_cursor->next(start_time_cursor);
+
+    if (ret == 0)
+        error_check(session->join(session, join_cursor, start_time_cursor, "compare=ge"));
+    else
+        return ret;
+
+    /* Find the ending record closest to desired end time. */
+    end_time_cursor->set_key(end_time_cursor, end_time);
+    error_check(end_time_cursor->search_near(end_time_cursor, &exact));
+    if (exact == 1)
+        ret = end_time_cursor->prev(end_time_cursor);
```

Comment:
If ret here is an internal failure, we are still ignoring and discarding it. Since the return gets forced into a boolean. Talk to me if you need guidance on how to fix this.